crates/op-rbuilder/src/tests/vanilla/revert.rs

```diff
@@ -123,42 +124,106 @@ async fn revert_protection_bundle() -> eyre::Result<()> {
-    assert!(block_generated.not_includes(*reverted_bundle.tx_hash()));
+    assert!(block_generated.includes(*reverted_bundle_0.tx_hash()));
+
+    let reverted_bundle_1 = harness
+        .create_transaction()
+        .with_revert()
+        .with_bundle(bundle_opts)
+        .send()
+        .await?;
+
+    // Test 3: Bundle reverts. It is not included in the block since it reverts
+    // and the hash is not in the reverted_hashes field.
+    let block_generated = generator.generate_block().await?; // Block 4
+    assert!(block_generated.not_includes(*reverted_bundle_1.tx_hash()));
 
     // After the block the transaction is still pending in the pool
     assert!(harness
-        .check_tx_in_pool(*reverted_bundle.tx_hash())
+        .check_tx_in_pool(*reverted_bundle_1.tx_hash())
         .await?
         .is_pending());
 
     // Test 3: Chain progresses beyond the bundle range. The transaction is dropped from the pool
-    generator.generate_block().await?; // Block 4
+    generator.generate_block().await?; // Block 5
     assert!(harness
-        .check_tx_in_pool(*reverted_bundle.tx_hash())
+        .check_tx_in_pool(*reverted_bundle_1.tx_hash())
         .await?
         .is_pending());
 
-    generator.generate_block().await?; // Block 5
+    generator.generate_block().await?; // Block 6
     assert!(harness
-        .check_tx_in_pool(*reverted_bundle.tx_hash())
+        .check_tx_in_pool(*reverted_bundle_1.tx_hash())
         .await?
         .is_dropped());
 
     Ok(())
 }
 
+/// Test the behaviour of the revert protection bundle with a min block number.
+#[tokio::test]
+async fn revert_protection_bundle_min_block_number() -> eyre::Result<()> {
+    let harness = TestHarnessBuilder::new("revert_protection_bundle_min_block_number")
+        .with_revert_protection()
+        .build()
+        .await?;
+
+    let mut generator = harness.block_generator().await?;
+
+    // The bundle is valid when the min block number is equal to the current block
+    let bundle_with_min_block = harness
+        .create_transaction()
+        .with_revert() // the transaction reverts but it is included in the block
+        .with_reverted_hash()
+        .with_bundle(BundleOpts {
+            block_number_max: None,
+            block_number_min: Some(2),
+        })
+        .send()
+        .await?;
+
+    let block = generator.generate_block().await?; // Block 1, bundle still not valid
+    assert!(block.not_includes(*bundle_with_min_block.tx_hash()));
+
+    let block = generator.generate_block().await?; // Block 2, bundle is valid
+    assert!(block.includes(*bundle_with_min_block.tx_hash()));
+
+    // Send a bundle with a match of min and max block number
+    let bundle_with_min_and_max_block = harness
+        .create_transaction()
+        .with_revert()
+        .with_reverted_hash()
+        .with_bundle(BundleOpts {
+            block_number_max: Some(4),
+            block_number_min: Some(4),
```

Comment:
very opinionated nit: maybe `Range<BlockNumber>` would make it more expressive because users will be able to specify it in various ways such as `a..b`, `a..`, `a..=b`, `..b`, `..=b`, etc.